From 3795398009049302a7b306d96a8e310163d60bb6 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@kde.org>
Date: Fri, 16 Oct 2026 13:39:41 +0200
Subject: [PATCH 1/3] workspace: turn screens off before suspend (again)

This was moved to powerdevil with 500b796542e9d2cb6fc415f6df4a8af1f4035fd4, but
if powerdevil crashes or doesn't inhibit suspend before requesting dpms off,
it's possible for the screen to stay on and for us to leak screen contents when
we lock the screen before suspend.

To make sure screen locking is secured against such cases, simply turn screens
off in KWin again.
---
 src/workspace.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/workspace.cpp b/src/workspace.cpp
index 95564ef51fd7..b641686a4e0a 100644
--- a/src/workspace.cpp
+++ b/src/workspace.cpp
@@ -367,6 +367,10 @@ void Workspace::init()
     connect(kwinApp()->outputBackend(), &OutputBackend::outputAdded, this, [this](BackendOutput *output) {
         connect(output, &BackendOutput::dpmsModeChanged, this, &Workspace::maybeUpdateDpmsState);
     });
+
+    connect(kwinApp()->session(), &Session::aboutToSleep, this, [this]() {
+        requestDpmsState(DpmsState::Off);
+    });
 }
 
 QString Workspace::outputLayoutId() const
-- 
2.55.0

